lib/pure/includes/osenv.nim

```diff
@@ -175,6 +181,32 @@ proc putEnv*(key, val: string) {.tags: [WriteEnvEffect].} =
       if c_putenv(environment[indx]) != 0'i32:
         raiseOSError(osLastError())
 
+proc unsetEnv*(key: string) {.tags: [WriteEnvEffect].} =
+  ## Removes the `environment variable`:idx: named `key`.
+  ## If an error occurs, `OSError` is raised.
+  ##
+  ## See also:ven
+  ## * `getEnv proc <#getEnv,string,string>`_
+  ## * `existsEnv proc <#existsEnv,string>`_
+  ## * `putEnv proc <#putEnv,string,string>`_
+  ## * `envPairs iterator <#envPairs.i>`_
+  when nimvm:
+    discard "built into the compiler"
+  else:
+    var indx = findEnvVar(key)
+    if indx < 0: return # Do nothing if the env var is not already set
+    let keyToRemove = environment[indx].split('=')[0] # "KEY=VAL" -> "KEY"; assumes that KEY does not have '=' char.
```

Comment:
Why not use ``let idx = find(..., '=')`` so that VAL can contain a '='?

Reply:
Right now, VAL is allowed to contain '='.

I am doing a left split and picking the first element.

Reply:
While writing above code, I was wondering if the `environment` should be a seq of tuples where the tuple has `key` and `val` fields.

Then we don't have to do poorman's parsing using the `=` char.